projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java

```diff
@@ -452,6 +452,19 @@ public Optional<JobDeploymentStatus> readCronJob(String cronJobName) {
     return jobStatus.isPresent() ? jobStatus : readV1CronJob(cronJobName);
   }
 
+  public boolean cronJobExists(String cronJobName) {
+    try {
+      initBatchV1beta1Api().readNamespacedCronJob(cronJobName, namespace, null);
+    } catch (ApiException e) {
+      try {
+        initBatchV1Api().readNamespacedCronJob(cronJobName, namespace, null);
+      } catch (ApiException e2) {
+        return false;
+      }
+    }
```

Comment:
add logging (it may be debug) or we will wonder if there's an issue in the future . 


It's good idea when logging or re-throwing to wrap it in our KubernetesException since it prints more details (by default ApiException does not print http body where the actual reason for the error is)

Reply:
Should we not return false only if the API call returns 404 Not Found. If it's 5xx error, we should re-throw. Otherwise the logic would be faulty, we will think a corn job does not exists while there is network or other infrastructure issue.

Reply:
This is a really good point. This is actually a bug in the existing code and is not introduced in this PR. 
But I will address it.